From: Sam Bradshaw <sbradshaw@micron.com>
Date: Wed, 15 May 2013 10:04:34 +0200
Subject: mtip32xx: Fix NULL pointer dereference during module unload
Git-repo: http://git.kernel.org/pub/scm/linux/kernel/git/axboe/linux-block.git
Git-commit: 974a51a245c2c8bece21cf2d3cbfc8261260f729
Patch-mainline: Queued in subsystem maintainer repo
References: FATE#311167,bug#820569

An open file-handle to one or more of the driver exported debugfs
nodes causes raciness in recursive removal during module unload;
sometimes a stale parent dentry is dereferenced when more than 1
pci device is present.

Signed-off-by: Sam Bradshaw <sbradshaw@micron.com>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Asai Thambi S P <asamymuthupa@micron.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---

 drivers/block/mtip32xx/mtip32xx.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/block/mtip32xx/mtip32xx.c
+++ b/drivers/block/mtip32xx/mtip32xx.c
@@ -3002,7 +3002,8 @@ static int mtip_hw_debugfs_init(struct driver_data *dd)
 
 static void mtip_hw_debugfs_exit(struct driver_data *dd)
 {
-	debugfs_remove_recursive(dd->dfs_node);
+	if (dd->dfs_node)
+		debugfs_remove_recursive(dd->dfs_node);
 }
 
 


